interpreter/misc_test.go

```diff
@@ -598,7 +598,7 @@ func TestInterpretInvalidArrayIndexing(t *testing.T) {
 			_, err := inter.Invoke("test", indexValue)
 			RequireError(t, err)
 
-			var indexErr interpreter.ArrayIndexOutOfBoundsError
+			var indexErr *interpreter.ArrayIndexOutOfBoundsError
```

Comment:
The location range of the `ArrayIndexOutOfBoundsError`  now matches to that of interpreter, in most cases. 
(Except for certain cases; see the below test for example, and the explanation)